hexa/connections/schema/types.py

```diff
@@ -0,0 +1,31 @@
+import logging
+
+from ariadne import ObjectType
+from openhexa.toolbox.dhis2.api import DHIS2Error
+
+from hexa.connections.dhis2_client_helper import get_dhis2_metadata
+
+dhis2_connection = ObjectType("DHIS2Connection")
+
+
+@dhis2_connection.field("query")
+def resolve_query(dhis2_client, info, **kwargs):
+    try:
+        metadata = get_dhis2_metadata(
+            dhis2_client,
+            type=kwargs.get("type"),
+            fields=kwargs.get("fields", "id,name"),
+            filter=kwargs.get("filter"),
+        )
+
+        result = [{"id": item.get("id"), "name": item.get("name")} for item in metadata]
```

Comment:
The result is always ids and names even though fields can be any attributes ?

Reply:
Yeah, we only need id and name for the feature, so i did limit the result with id name as metadataResult. To make it explicit I could name it metadataQuery, do you think it will be more clear?

Reply:
I don't tink the name is a problem.

But it means that `fields` parameter has no effect at all on the result of the query since we always return id and name ? 

Also if id and name are not in the `fields`, the result will be `None` for all items. 

I think that either we make it generic and support `fields` or we remove it to avoid issues like the one described above

Reply:
i agree that in scope of the current feature it seems contradictory to offer the fields parameter and in the same time to restrict the result to fields and name. I will lock it to id,name to fetch metadata